lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -409,6 +409,155 @@ def test_kwargs(self):
         self.assertMaskedArrayEqual(expected_result, res_cube.data)
 
 
+class TestSlices(tests.IrisTest):
+    '''
+    This class tests the capability of iris.cube.Cube.slices(), including its
+    ability to correctly re-order the dimensions.
+    '''
+    def setUp(self):
+        '''
+        setup a 4D iris cube, each dimension is length 1.
+        The dimensions are;
+            dim1: height
+            dim2: time
+            dim3: longitude
+            dim4: latitude
+        '''
+        self.cube = iris.cube.Cube(np.array([[[[8.]]]]))
+        self.cube.add_dim_coord(iris.coords.DimCoord([0], "height"), [0])
+        self.cube.add_dim_coord(iris.coords.DimCoord([0], "time"), [1])
+        self.cube.add_dim_coord(iris.coords.DimCoord([0], "longitude"), [2])
+        self.cube.add_dim_coord(iris.coords.DimCoord([0], "latitude"), [3])
+
+    @staticmethod
+    def expected_cube_setup(dim1name, dim2name, dim3name):
+        '''
+        input:
+        ------
+            dim1name: str
+                name of the first dimension coordinate
+            dim2name: str
+                name of the second dimension coordinate
+            dim3name: str
+                name of the third dimension coordinate
+        output:
+        ------
+            cube: iris cube
+                iris cube with the specified axis holding the data 8
+        '''
+        cube = iris.cube.Cube(np.array([[[8.]]]))
+        cube.add_dim_coord(iris.coords.DimCoord([0], dim1name), [0])
+        cube.add_dim_coord(iris.coords.DimCoord([0], dim2name), [1])
+        cube.add_dim_coord(iris.coords.DimCoord([0], dim3name), [2])
+        return cube
+
+    def make_the_test(self, dim1, dim2, dim3, dim_to_remove):
+        '''
+        does two things:
+        (1) slices the 4D cube in dim1, dim2, dim3 (and removes the scalar
+        coordinate) and
+        (2) sets up a 3D cube with dim1, dim2, dim3.
+        input:
+        -----
+            dim1: str
+                name of first dimension
+            dim2: str
+                name of second dimension
+            dim3: str
+                name of third dimension
+            dim_to_remove: str
+                name of the dimension that transforms into a scalar coordinate
+                when slicing the cube.
+        output:
+        ------
+            sliced_cube: 3D cube
+                the cube that results if slicing the original cube
+            expected_cube: 3D cube
+                a cube set up with the axis corresponding to the dims
+        '''
+        sliced_cube = next(self.cube.slices([dim1, dim2, dim3]))
+        sliced_cube.remove_coord(dim_to_remove)
+        expected_cube = self.expected_cube_setup(dim1, dim2, dim3)
+        return sliced_cube, expected_cube
+
+    def test_height_long_lat(self):
```

Comment:
I've had a hard time trying to work out whether there is some special logic in the cases you test here (there are 7, out of a possible 4 * 3 * 2 = 24, I think ??).

This may sound over-clever, but if you are testing a whole bunch of cases then why not just have code to generate all of them??
If any ordering of the 4 coords will make a possible testcase, then you can just use `itertools.permutations()`.

If you do that, you probably have to do all the assertions within one testcase routine, so you should ensure that if one of them fails the resulting message makes it quite clear which one it was.